pcre2-sys/build.rs

```diff
@@ -173,6 +173,7 @@ fn pcre2_sys_static() -> Option<bool> {
 // propagating to all `aarch64`-based targets and the `x86_64` equivalents.
 fn enable_jit(target: &str, builder: &mut cc::Build) {
     if !target.starts_with("aarch64-apple") &&
+        !(target.starts_with("aarch64") && target.ends_with("musl")) &&
```

Comment:
OK so this function is not written in an ideal style. And this addition here does not seem to be in the same category of "apple" targets, right?

I would like to see it written this way instead:

```rust
fn enable_jit(target: &str, builder: &mut cc::Build) {
    if target.starts_with("aarch64-apple") || ... {
        return;
    }
    if target.starts_with("aarch64") && target.ends_with("musl") {
        return;
    }
    builder.define("SUPPORT_JIT", "1");
}
```

And the comment on `enable_jit` currently should be moved just above the first `if` statement and a _new_ comment that you'll need to write explaining in more detail why `aarch64*musl` is being skipped should be written above the second `if` statement.

Reply:
:+1: 

The error we have been running into is

```
ld.lld: error: undefined symbol: __clear_cache
          >>> referenced by sljitNativeARM_64.c:333 (pcre2/src/sljit/sljitNativeARM_64.c:333)
          >>>               pcre2_jit_compile.o:(jit_compile) in archive /build/target/aarch64-unknown-linux-musl/release/deps/libpcre2_sys-85eff8d10f292f3e.rlib
          clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

It appears to be a recurring issue for sljit based on https://github.com/zherczeg/sljit/search?q=clear_cache&type=commits